cider/exec/template/ArithmeticIR.cpp

```diff
@@ -117,61 +119,105 @@ std::unique_ptr<CodegenColValues> CodeGenerator::codegenArithFun(
 
   CHECK(lhs_fixsize && rhs_fixsize);
 
-  if (overflow_check) {
-    // TODO: overflow_check code generation.
-    CIDER_THROW(CiderUnsupportedException, "");
-  }
-
   llvm::Value* value = nullptr;
   auto lh_value = lhs_fixsize->getValue(), rh_value = rhs_fixsize->getValue();
+
+  if (!needs_error_check) {
+    switch (bin_oper->get_optype()) {
+      case kMINUS:
+        value = lh_value->getType()->isIntegerTy()
+                    ? cgen_state_->ir_builder_.CreateSub(lh_value, rh_value)
+                    : cgen_state_->ir_builder_.CreateFSub(lh_value, rh_value);
+        break;
+      case kPLUS:
+        value = lh_value->getType()->isIntegerTy()
+                    ? cgen_state_->ir_builder_.CreateAdd(lh_value, rh_value)
+                    : cgen_state_->ir_builder_.CreateFAdd(lh_value, rh_value);
+        break;
+      case kMULTIPLY:
+        value = lh_value->getType()->isIntegerTy()
+                    ? cgen_state_->ir_builder_.CreateMul(lh_value, rh_value)
+                    : cgen_state_->ir_builder_.CreateFMul(lh_value, rh_value);
+        break;
+      case kDIVIDE:
+        value = lh_value->getType()->isIntegerTy()
+                    ? cgen_state_->ir_builder_.CreateSDiv(lh_value, rh_value)
+                    : cgen_state_->ir_builder_.CreateFDiv(lh_value, rh_value);
+        break;
+      case kMODULO:
+        value = lh_value->getType()->isIntegerTy()
+                    ? cgen_state_->ir_builder_.CreateSRem(lh_value, rh_value)
+                    : cgen_state_->ir_builder_.CreateFRem(lh_value, rh_value);
+        break;
+      default:
+        CHECK(false);
+    }
+    return std::make_unique<FixedSizeColValues>(value, null);
+  }
+  const auto lhs_operand = bin_oper->get_left_operand();
+  const auto rhs_operand = bin_oper->get_right_operand();
+  const auto& lhs_type = lhs_operand->get_type_info();
+  const auto& rhs_type = rhs_operand->get_type_info();
+  const auto int_typename = numeric_or_time_interval_type_name(lhs_type, rhs_type);
+  const auto null_check_suffix = get_null_check_suffix(lhs_type, rhs_type);
+  const auto& oper_type = rhs_type.is_timeinterval() ? rhs_type : lhs_type;
   switch (bin_oper->get_optype()) {
     case kMINUS:
-      value = lh_value->getType()->isIntegerTy()
-                  ? cgen_state_->ir_builder_.CreateSub(lh_value, rh_value)
-                  : cgen_state_->ir_builder_.CreateFSub(lh_value, rh_value);
-      break;
     case kPLUS:
-      value = lh_value->getType()->isIntegerTy()
-                  ? cgen_state_->ir_builder_.CreateAdd(lh_value, rh_value)
-                  : cgen_state_->ir_builder_.CreateFAdd(lh_value, rh_value);
-      break;
-    case kMULTIPLY:
-      value = lh_value->getType()->isIntegerTy()
-                  ? cgen_state_->ir_builder_.CreateMul(lh_value, rh_value)
-                  : cgen_state_->ir_builder_.CreateFMul(lh_value, rh_value);
-      break;
+    case kMULTIPLY: {
+      // add overflow check for only INT-like types?
```

Comment:
https://releases.llvm.org/2.6/docs/LangRef.html#int_overflow. LLVM only provide integer overflow check, I think this is because  floating-point return result will become an infinity rather than wrapping around like integer.